File: frontend/types/payslip.ts
/**
 * 급여명세서 관련 타입 정의
 */

// 발송 방법 타입
export type SendMethod = 'email' | 'kakao' | 'both';

// 발송 상태 타입
export type SendStatus = 'pending' | 'sent' | 'failed';

// 급여명세서 엔티티
export interface Payslip {
  id: string;
  employee_id: string;
  employee_name: string;
  company_name: string;
  year: number;
  month: number;
  payment_date: string | null;

  // 지급 항목
  base_salary: number;
  weekly_allowance: number;
  overtime_pay: number;
  night_pay: number;
  holiday_pay: number;
  meal_allowance: number;
  transport_allowance: number;
  total_payment: number;

  // 공제 항목
  national_pension: number;
  health_insurance: number;
  long_term_care: number;
  employment_insurance: number;
  income_tax: number;
  local_income_tax: number;
  total_deduction: number;

  // 실수령액
  net_salary: number;

  // 발송 정보
  send_status: SendStatus;
  sent_at: string | null;
  sent_via: string | null;
  created_at: string;
}

// 급여명세서 생성 요청
export interface CreatePayslipRequest {
  employee_id: string;
  year: number;
  month: number;
  payment_date: string;
  base_salary: number;
  weekly_allowance?: number;
  overtime_pay?: number;
  night_pay?: number;
  holiday_pay?: number;
  meal_allowance?: number;
  transport_allowance?: number;
  national_pension?: number;
  health_insurance?: number;
  long_term_care?: number;
  employment_insurance?: number;
  income_tax?: number;
  local_income_tax?: number;
}

// 급여명세서 발송 요청
export interface SendPayslipRequest {
  method: SendMethod;
  email?: string;
}

/**
 * 금액을 한국 원화 형식으로 포맷
 * @param amount 포맷할 금액 (숫자)
 * @returns 콤마 구분 + '원' 단위 문자열 (예: 3,000,000원)
 */
export function formatCurrency(amount: number): string {
  return new Intl.NumberFormat('ko-KR').format(amount) + '원';
}

/**
 * 발송 상태 한국어 라벨 반환
 * @param status 발송 상태
 * @returns 한국어 라벨 문자열
 */
export function getSendStatusLabel(status: SendStatus): string {
  const labels: Record<SendStatus, string> = {
    pending: '미발송',
    sent: '발송완료',
    failed: '발송실패',
  };
  return labels[status];
}

/**
 * 발송 상태에 따른 Tailwind CSS 텍스트 색상 클래스 반환
 * @param status 발송 상태
 * @returns Tailwind CSS 클래스 문자열
 */
export function getSendStatusColor(status: SendStatus): string {
  const colors: Record<SendStatus, string> = {
    pending: 'text-slate-500',
    sent: 'text-green-600',
    failed: 'text-red-600',
  };
  return colors[status];
}

File: backend/app/schemas/payslip.py
# Payslip 관련 스키마
from datetime import datetime, date
from typing import Literal
from pydantic import BaseModel, Field
from decimal import Decimal
import uuid


class PayslipCreate(BaseModel):
    """급여명세서 생성 요청"""
    employee_id: uuid.UUID
    year: int = Field(..., ge=2020, le=2100)
    month: int = Field(..., ge=1, le=12)
    payment_date: date
    # 지급 항목
    base_salary: Decimal = Field(..., ge=0)
    weekly_allowance: Decimal = Field(default=Decimal("0"), ge=0)
    overtime_pay: Decimal = Field(default=Decimal("0"), ge=0)
    night_pay: Decimal = Field(default=Decimal("0"), ge=0)
    holiday_pay: Decimal = Field(default=Decimal("0"), ge=0)
    meal_allowance: Decimal = Field(default=Decimal("0"), ge=0)
    transport_allowance: Decimal = Field(default=Decimal("0"), ge=0)
    # 공제 항목
    national_pension: Decimal = Field(default=Decimal("0"), ge=0)
    health_insurance: Decimal = Field(default=Decimal("0"), ge=0)
    long_term_care: Decimal = Field(default=Decimal("0"), ge=0)
    employment_insurance: Decimal = Field(default=Decimal("0"), ge=0)
    income_tax: Decimal = Field(default=Decimal("0"), ge=0)
    local_income_tax: Decimal = Field(default=Decimal("0"), ge=0)


class PayslipResponse(BaseModel):
    """급여명세서 응답"""
    id: uuid.UUID
    employee_id: uuid.UUID
    employee_name: str
    company_name: str
    year: int
    month: int
    payment_date: date | None = None
    # 지급 항목
    base_salary: Decimal
    weekly_allowance: Decimal
    overtime_pay: Decimal
    night_pay: Decimal
    holiday_pay: Decimal
    meal_allowance: Decimal
    transport_allowance: Decimal
    total_payment: Decimal
    # 공제 항목
    national_pension: Decimal
    health_insurance: Decimal
    long_term_care: Decimal
    employment_insurance: Decimal
    income_tax: Decimal
    local_income_tax: Decimal
    total_deduction: Decimal
    # 실수령액
    net_salary: Decimal
    # 발송 상태
    send_status: str
    sent_at: datetime | None = None
    sent_via: str | None = None
    created_at: datetime

    class Config:
        from_attributes = True


class PayslipListResponse(BaseModel):
    """급여명세서 목록 응답"""
    id: uuid.UUID
    employee_id: uuid.UUID
    employee_name: str
    year: int
    month: int
    net_salary: Decimal
    send_status: str
    created_at: datetime


class SendPayslipRequest(BaseModel):
    """급여명세서 발송 요청"""
    method: Literal["email", "kakao", "both"] = "email"
    email: str | None = None
